mapper/javabean/src/main/java/org/hibernate/search/mapper/javabean/model/impl/JavaBeanBootstrapIntrospector.java

```diff
@@ -87,7 +87,7 @@ public JavaBeanBootstrapIntrospector(MethodHandles.Lookup lookup) {
 		return annotationHelper.getAnnotationsByMetaAnnotationType( annotatedElement, metaAnnotationType );
 	}
 
-	PropertyHandle<?> createPropertyHandle(String name, Method method) throws IllegalAccessException {
+	<T> PropertyHandle<T> createPropertyHandle(String name, Method method) throws IllegalAccessException {
```

Comment:
As I mentioned in a previous comment, this is not a good idea. Here you're saying "the factory can provide the caller with a `PropertyHandle<T>`, and you can choose `T`, regardless of the `Method` passed". Which is wrong: `T` is predefined, depends on the `Method` passed, and cannot be chosen by the caller.

So this should really be:

```
PropertyHandle<?> createPropertyHandle(String name, Method method) throws IllegalAccessException {
```

Which is what you had before.

I know what you wanted to do was to solve another problem; I'll add a comment there.